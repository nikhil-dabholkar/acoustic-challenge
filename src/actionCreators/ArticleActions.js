import { LOAD_ARTICLE_DETAILS_DATA, LOAD_ALL_ARTICLES_DATA } from '../constants/constants';
const axios = require('axios');

export const fetchSelectedArticleData = (articleId) => {
    return async (dispatch) => {
        const response = await axios.get(`https://content-eu-4.content-cms.com/api/859f2008-a40a-4b92-afd0-24bb44d10124/delivery/v1/content/${articleId}`);
        console.log(response);
        dispatch({
            type: LOAD_ARTICLE_DETAILS_DATA,
            payload: response.data
        })
    }
}

export const fetchAllArticlesData = () => {
    return async (dispatch) => {
        const response = await axios.get(`https://run.mocky.io/v3/8421cb68-a9d9-4575-ad5d-b4396c864231/`);
        console.log(response);
        dispatch({
            type: LOAD_ALL_ARTICLES_DATA,
            payload: response.data
        })
    }
}
