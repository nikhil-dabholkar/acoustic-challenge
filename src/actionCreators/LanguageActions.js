import { SELECT_LANGUAGE_PREFERENCE } from '../constants/constants'

/**
 * This action will be called when user changes the language
 */
export const switchLanguagePreference = (payload) => ({
  type: SELECT_LANGUAGE_PREFERENCE,
  payload: payload
})
