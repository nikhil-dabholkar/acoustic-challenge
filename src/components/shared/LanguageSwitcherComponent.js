import React from 'react';
import { Link } from 'carbon-components-react';

function LanguageSwitcherComponent({availableLanguages, selectedLanguage, switchLanguage}) {
    return (<React.Fragment>
        {availableLanguages.map((language) => {
            if (language.locale !== selectedLanguage.locale) {
                return (<span key={language.locale + language.region} >
                    <Link href="#" onClick={() => switchLanguage(language)}>{language.abbr}</Link>
                </span>)
            }
            return <span key={language.locale + language.region} >{language.abbr}</span>
        })}
    </React.Fragment>);
}

export default LanguageSwitcherComponent;
