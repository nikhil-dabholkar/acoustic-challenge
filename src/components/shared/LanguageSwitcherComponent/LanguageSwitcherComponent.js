import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'carbon-components-react';

function LanguageSwitcherComponent({ availableLanguages, selectedLanguage, switchLanguage }) {
    return (<React.Fragment>
        <div data-testid="languageSwitcherComponent">
            {availableLanguages.map((language, index, array) => {
                const delimitter = " | ";
                if (language.locale !== selectedLanguage.locale) {
                    return (<span key={language.locale + language.region} >
                        <Link href="#" onClick={() => switchLanguage(language)}>{language.label}</Link>
                        {(index < array.length - 1) ? delimitter : ""}
                    </span>)
                }
                return (<span key={language.locale + language.region} >{language.label}{(index < array.length - 1) ? delimitter : ""}</span>)
            })}
        </div>
    </React.Fragment>);
}

LanguageSwitcherComponent.propTypes = {
    switchLanguage: PropTypes.func.isRequired,
    selectedLanguage: PropTypes.object.isRequired,
    availableLanguages: PropTypes.array.isRequired
}

export default React.memo(LanguageSwitcherComponent);
