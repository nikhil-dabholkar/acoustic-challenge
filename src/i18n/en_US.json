{
    "app.header": "English",
    "detailsComponent.dateTooltip": "Date and time is used when ordering a list of design articles. The time will not display in the website.",
    "detailsComponent.bodySectionTooltip": "The design topic, when displayed in the website, will interleave the text chunks and images to create a fluid layout.",
    "detailsComponent.topicTitle": "Topic title",
    "detailsComponent.authorName": "Author name",
    "detailsComponent.date": "Date",
    "detailsComponent.leadImage": "Lead image",
    "detailsComponent.leadImageCaption": "Lead image caption",
    "detailsComponent.LeadImageCredit": "Lead image credit",
    "detailsComponent.profiles": "Profiles",
    "detailsComponent.lead": "Lead",
    "detailsComponent.card": "Card",
    "detailsComponent.default": "Default",
    "detailsComponent.textForBodySection": "Text for body section",
    "listComponent.id": "Id",
    "listComponent.name": "Name",
    "headerComponent.home": "Home",
    "headerComponent.articles": "Articles"
}