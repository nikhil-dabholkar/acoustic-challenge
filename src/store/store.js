import rootReducer from '../reducer/index';
import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';

export const store = createStore(rootReducer, applyMiddleware(thunk))