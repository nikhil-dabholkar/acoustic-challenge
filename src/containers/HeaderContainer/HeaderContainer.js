import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { switchLanguagePreference } from '../../actionCreators/LanguageActions';
import { HeaderComponent } from '../../components/index';

function HeaderContainer({ language, switchLanguage }) {
    return (<React.Fragment>
        <div data-testid="headerContainer"></div>
        <HeaderComponent language={language} switchLanguage={switchLanguage}></HeaderComponent>
    </React.Fragment>
    );
}

const mapStateToProps = state => {
    return ({
        language: state.languageReducer
    })
};

const mapDispatchToProps = {
    switchLanguage: switchLanguagePreference
}

HeaderContainer.propTypes = {
    language: PropTypes.object.isRequired,
    switchLanguage: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(React.memo(HeaderContainer));
