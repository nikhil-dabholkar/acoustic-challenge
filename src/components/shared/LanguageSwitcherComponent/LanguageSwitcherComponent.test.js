import React from 'react';
import HeaderContainer from '../../../containers/HeaderContainer/HeaderContainer';
import LanguageSwitcherComponent from './LanguageSwitcherComponent';
import { IntlProvider } from 'react-intl';
import { store } from "../../../store/store";
import { Provider } from 'react-redux';
import { render, cleanup, screen } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import en_US from "../../../i18n/en_US.json";
import { switchLanguagePreference } from "../../../actionCreators/LanguageActions";

afterEach(cleanup);

function renderWithRouter(
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
    return {
        ...render(<Router history={history}>{ui}</Router>),
        history,
    }
}

function rendererWithRouter(
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
    return {
        ...renderer.create(<Router history={history}>{ui}</Router>),
        history,
    }
}

describe('LanguageSwitcherComponent', () => {
    test('Check if LanguageSwitcherComponent Renders', async () => {
        const route = '/home';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("languageSwitcherComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test('Check if LanguageSwitcherComponent Renders', async () => {
        const route = '/articles';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("languageSwitcherComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test('Check if LanguageSwitcherComponent Renders', async () => {
        const route = '/articles/fa9519d5-0363-4b8d-8e1f-627d802c08a8';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("languageSwitcherComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test('Check if LanguageSwitcherComponent Renders', async () => {
        const route = '/articles/invalid_item';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("languageSwitcherComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    test('Check if LanguageSwitcherComponent Renders', async () => {
        const route = '/';
        renderWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><HeaderContainer></HeaderContainer></IntlProvider></Provider>, { route })
        const lazyElement = await screen.findByTestId("languageSwitcherComponent")
        expect(lazyElement).toBeInTheDocument()
    })

    it("Check if matches LanguageSwitcherComponent Snapshot", async () => {
        const route = '/';
        const availableLanguages = [{"abbr": "EN", "locale": "en", "label": "English", "region": "US"},{"abbr": "PL", "locale": "pl", "label": "Polish", "region": "PL"}];
        const selectedLanguage = {"abbr": "EN", "locale": "en", "label": "English", "region": "US"};
        const tree = rendererWithRouter(<Provider store={store}><IntlProvider locale="en" messages={en_US}><LanguageSwitcherComponent availableLanguages={availableLanguages} selectedLanguage={selectedLanguage} switchLanguage={switchLanguagePreference}></LanguageSwitcherComponent></IntlProvider></Provider>, { route }).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
