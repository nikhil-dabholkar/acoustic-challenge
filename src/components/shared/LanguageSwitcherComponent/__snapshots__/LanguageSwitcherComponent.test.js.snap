// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcherComponent Check if matches LanguageSwitcherComponent Snapshot 1`] = `
<div
  data-testid="languageSwitcherComponent"
>
  <span>
    English
     | 
  </span>
  <span>
    <a
      className="bx--link"
      href="#"
      onClick={[Function]}
    >
      Polish
    </a>
    
  </span>
</div>
`;
